src/main/java/org/jabref/logic/integrity/TitleChecker.java

```diff
@@ -48,11 +49,16 @@ public TitleChecker(BibDatabaseContext databaseContext) {
             valueOnlySpacesWithinCurlyBraces = matcher.replaceAll("");
         }
 
-        boolean hasCapitalLettersThatBibtexWillConvertToSmallerOnes = HAS_CAPITAL_LETTERS
-                .test(valueOnlySpacesWithinCurlyBraces);
-
-        if (hasCapitalLettersThatBibtexWillConvertToSmallerOnes) {
-            return Optional.of(Localization.lang("capital letters are not masked using curly brackets {}"));
+        // Delimiters are . ! ? ; :
+        String[] splitTitle = valueOnlySpacesWithinCurlyBraces.split(DELIMITERS);
+        for (String subTitle : splitTitle) {
+            subTitle = subTitle.trim();
+            if (subTitle.length() > 0) {
+                subTitle = subTitle.substring(1);
+            }
+            if (HAS_CAPITAL_LETTERS.test(subTitle)) {
```

Comment:
Please move this before the `}`. In this case, `subTitle` could be `0` characters long and thus no test is necessary